tl-services/src/main/java/org/egov/tl/service/EnrichmentService.java

```diff
@@ -457,6 +459,14 @@ private void setLicenseNumberAndIssueDate(TradeLicenseRequest request,List<Strin
                     Long time = System.currentTimeMillis();
                     license.setIssuedDate(time);
                     license.setValidFrom(time);
+                    if (mdmsData != null && businessService.equalsIgnoreCase(businessService_BPA)) {
```

Comment:
please move the logic to enrichTLCreateRequest() function where we are enriching validTo

Reply:
Hi Aniket, validTo should be calculated after generating the licenseNumber with the help of validityPeriod for a given tradeType. and license Number will be generated in update call. 
enrichTLCreateRequest will be called in create. Please suggest if am missing any thing obvious here.